homeassistant/components/influxdb.py

```diff
@@ -120,7 +120,11 @@ def influx_event_listener(event):
 
         for key, value in state.attributes.items():
             if key != 'unit_of_measurement':
-                json_body[0]['fields'][key] = value
+                if isinstance(value, (str, float, bool)):
+                    json_body[0]['fields'][key] = value
+                elif isinstance(value, int):
```

Comment:
Agree about overhaul needed, but I think this type of change would have been suited to go in the PR that should come out of #4696.